components/form.js

```diff
@@ -617,34 +691,34 @@ function InputInner ({
 }
 
 const INITIAL_SUGGESTIONS = { array: [], index: 0 }
-export function UserSuggest ({
-  query, onSelect, dropdownStyle, children,
-  transformUser = user => user, selectWithTab = true, filterUsers = () => true
+
+export function BaseSuggest ({
+  query, onSelect, dropdownStyle,
+  transformItem = item => item, selectWithTab = true, filterItems = () => true,
+  getSuggestionsQuery, queryName, itemsField,
+  children
 }) {
-  const [getSuggestions] = useLazyQuery(USER_SUGGESTIONS, {
+  const [getSuggestions] = useLazyQuery(getSuggestionsQuery, {
     onCompleted: data => {
       query !== undefined && setSuggestions({
-        array: data.userSuggestions
-          .filter((...args) => filterUsers(query, ...args))
-          .map(transformUser),
+        array: data[itemsField]
+          .filter((...args) => filterItems(query, ...args))
+          .map(transformItem),
         index: 0
       })
     }
   })
-
   const [suggestions, setSuggestions] = useState(INITIAL_SUGGESTIONS)
   const resetSuggestions = useCallback(() => setSuggestions(INITIAL_SUGGESTIONS), [])
-
   useEffect(() => {
     if (query !== undefined) {
-      // remove both the leading @ and any @domain after nym
-      const q = query?.replace(/^[@ ]+|[ ]+$/g, '').replace(/@[^\s]*$/, '')
+      // remove the leading character and any trailing spaces
+      const q = query?.replace(/^[@ ~]+|[ ]+$/g, '').replace(/@[^\s]*$/, '').replace(/~[^\s]*$/, '')
```

Comment:
You had removed this trailing replacement. I'm not sure why it was there, but it's best if we keep it.